### PR TITLE
[basic] Replace 'could' and 'might'

### PR DESCRIPTION
**Pull request by jensmaurer**
as directed by ISO/CS.

Partially addresses #4319

### CODE REVIEW

File: source/basic.tex [hunk 1]
```diff
@@ -2965,7 +2965,7 @@
 a specialization of a template
 whose (possibly instantiated) declaration is an exposure.
 \begin{note}
-The specialization might have been implicitly or explicitly instantiated.
+A specialization can be produced by implicitly or explicitly instantiation.
```
Comment (tkoeppe) [inline source/basic.tex:2968]: Should these adverbs become adjectives?

File: source/basic.tex [hunk 2]
```diff
@@ -3442,7 +3442,7 @@
 lifetime ends there are significant restrictions on the use of the
 object, as described below, in~\ref{class.base.init} and
 in~\ref{class.cdtor}. Also, the behavior of an object under construction
-and destruction might not be the same as the behavior of an object whose
+and destruction is possibly not the same as the behavior of an object whose
```
Comment (tkoeppe) [inline source/basic.tex:3445]: This is very nice. We should give that as an example of possible rewordings.

File: source/basic.tex [hunk 3]
```diff
@@ -5262,8 +5261,8 @@
 \begin{note}
 A pointer past the end of an object\iref{expr.add}
 is not considered to point to an unrelated object
-of the object's type
-that might be located at that address.
+of the object's type,
+even if the unrelated object would be located at that address.
```
Comment (tkoeppe) [inline source/basic.tex:5265]: Can we just say "is located" here? What does the extra qualification achieve here?

File: source/basic.tex [hunk 4]
```diff
@@ -6351,7 +6350,7 @@
 guarantees, which in turn are stronger than weakly parallel forward progress
 guarantees.
 \begin{note}
-For example, some kinds of synchronization between threads of execution might only
+For example, some kinds of synchronization between threads of execution would only
 make progress if the respective threads of execution provide parallel forward progress
```
Comment (tkoeppe) [inline source/basic.tex:6354]: I feel like this should now say "if the respective threads ... _provided_ parallel forward progress guarantees." What do you think?

**Review by tkoeppe — CHANGES_REQUESTED** [review]

File: source/basic.tex (hunk 1 above)
Comment (jensmaurer) [reply source/basic.tex:2968]: fixed

File: source/basic.tex (hunk 3 above)
Comment (jensmaurer) [reply source/basic.tex:5265]: fixed

File: source/basic.tex (hunk 4 above)
Comment (jensmaurer) [reply source/basic.tex:6354]: of course
fixed

**Comment by jensmaurer** [comment]
fixed + force-pushed

File: source/basic.tex [hunk 5]
```diff
@@ -6277,7 +6276,7 @@
 \begin{example}
 A library I/O function that blocks until the I/O operation is complete can
 be considered to continuously check whether the operation is complete. Each
-such check might consist of one or more execution steps, for example using
+such check possibly consists of one or more execution steps, for example using
```
Comment (zygoloid) [inline source/basic.tex:6279]: What is the alternative? (If it doesn't consist of one or more execution steps, then what does it do?) Can we drop the "possibly"?

File: source/basic.tex [hunk 6]
```diff
@@ -6351,8 +6350,8 @@
 guarantees, which in turn are stronger than weakly parallel forward progress
 guarantees.
 \begin{note}
-For example, some kinds of synchronization between threads of execution might only
-make progress if the respective threads of execution provide parallel forward progress
+For example, some kinds of synchronization between threads of execution would only
+make progress if the respective threads of execution provided parallel forward progress
 guarantees, but will fail to make progress under weakly parallel guarantees.
```
Comment (zygoloid) [inline source/basic.tex:6355]: Is this really right? Should this instead say:
```suggestion
For example, some kinds of synchronization between threads of execution are only guaranteed to
make progress if the respective threads of execution provide parallel forward progress
guarantees, but will not necessarily make progress under weakly parallel guarantees.
```

File: source/basic.tex [hunk 7]
```diff
@@ -4225,7 +4225,7 @@
 When \tcode{D} is the type of a complete object, it will have a subobject of
 type \tcode{B}, so it must be aligned appropriately for a \tcode{long double}.
 If \tcode{D} appears as a subobject of another object that also has \tcode{B}
-as a virtual base class, the \tcode{B} subobject might be part of a different
+as a virtual base class, it is possible for the \tcode{B} subobject to be part of a different
 subobject, reducing the alignment requirements on the \tcode{D} subobject.
```
[the remark this answers is not in the document]
Comment (jensmaurer) [reply source/basic.tex:4229]: fixed

File: source/basic.tex (hunk 5 above)
Comment (jensmaurer) [reply source/basic.tex:6280]: dropped

**Comment by jensmaurer** [comment]
applied, squashed, and force-pushed

**Pull request merged**